src/distributed-transaction/pessimistic-and-optimistic-locking.md

```diff
@@ -0,0 +1,19 @@
+# Pessimistic & Optimistic locking
+
+There are mechanisms employed to manage the actions of multiple concurrent users on a database—the purpose is to prevent lost updates and dirty reads. The two types of locking are pessimistic locking and optimistic locking.
+
+## Pessimistic locking
+
+Pessimistic locking: a user who reads a record with the intention of updating it places an exclusive lock on the record to prevent other users from manipulating it. This means no one else can manipulate that record until the user releases the lock. The downside is that users can be locked out for a very long time, thereby slowing the overall system response and causing frustration.
```

Comment:
You don't need `Pessimistic locking:` since that's already the heading. Same with Optimistic.